packages/portal-contracts/src/Portal.sol

```diff
@@ -414,6 +421,42 @@ contract Portal is Owned {
         _transferToSender(tokToSend);
     }
 
+    // Update an existing bid or ask order. amountSats is negative if the order is an ask.
+    function updateOrder(uint256 orderID, int128 amountSats) public payable {
+        Order storage o = orderbook[orderID];
+
+        require(o.amountSats != 0, "Order not found");
+        require(o.maker == msg.sender, "Order not yours");
+
+        if (o.amountSats < 0) {
+          // This is an ask. Receive the ether to be sold and update the order.
+          require(amountSats < 0 && uint256(int256(-(o.amountSats + amountSats))) <= MAX_SATS, "Must be adding a valid amount of liquidity");
+
+          uint256 additionalValue = uint256(int256(-amountSats * int128(o.priceTokPerSat)));
+          o.amountSats += amountSats;
+
+          _transferFromSender(additionalValue);
+        } else {
+          // This is a bid, update the order and take the additional stake
+          require(amountSats > 0 && uint256(int256(o.amountSats)) + uint256(int256(amountSats)) <= MAX_SATS, "Must be adding a valid amount of liquidity");
+          uint256 expectedStakeTok = (uint256(int256(amountSats)) * uint256(o.priceTokPerSat) * stakePercent) / 100;
+          
+          // Update the order
+          o.amountSats += amountSats;
+          o.stakedTok += expectedStakeTok;
+
+          // Receive the additional stake
+          _transferFromSender(expectedStakeTok);
+        }
```

Comment:
I wonder if this is worth the complexity. 

The API, as it stands, is a big gnarly:
- For a bid, passing positive `amountSats` increases the size of your bid
- For an ask, passing positive `amountSats` *decreases* the size of your ask

In either case, we'd have to add a check to ensure bids are not turning into asks or vice versa.

I'm down to revisit later once we have testnet MMs. If it turns out that modifying bid *size* is a common need, we ship this.

My guess is that at least initially, immutable orders (that can only be cancelled and recreated) might be good enough.

Reply:
Ah I should probably have added more clarity:

This only lets you add additional liquidity, not remove it. The removing liquidity case strikes me as less likely, and therefore ok to go through the: cancel order -> place order flow.

Reply:
The adding liquidity case I think will be common. Lets take a WBTC <-> BTC portal where the spreads are tight and non-volatile. MM can have large liquidity, and most orders won't take the whole order from the consumer side. Being able to in place update your order to add additional liquidity strikes me as easier to manage than needing to place a separate order, which would then also fragment liquidity and make it harder to match orders.

Reply:
That being said, I don't think it'd be too hard to add later so we can punt on this for now